distributed/shuffle/_worker_extension.py

```diff
@@ -19,25 +23,32 @@
 from distributed.shuffle._arrow import (
     convert_partition,
     deserialize_schema,
-    dump_shards,
     list_of_buffers_to_table,
-    load_partition,
     serialize_table,
 )
 from distributed.shuffle._comms import CommShardsBuffer
 from distributed.shuffle._disk import DiskShardsBuffer
 from distributed.shuffle._limiter import ResourceLimiter
+from distributed.shuffle._rechunk import ChunkedAxes, NIndex
 from distributed.shuffle._shuffle import ShuffleId
 from distributed.sizeof import sizeof
 from distributed.utils import log_errors, sync
 
 if TYPE_CHECKING:
+    # TODO import from typing (requires Python >=3.10)
+    import numpy as np
     import pandas as pd
     import pyarrow as pa
+    from typing_extensions import TypeAlias
 
     from distributed.worker import Worker
 
+TransferShardIDType = TypeVar("TransferShardIDType")
+PartitionIDType = TypeVar("PartitionIDType")
+PartitionType = TypeVar("PartitionType")
```

Comment:
I suggest dropping the `Type` suffix or is this best practice? I think not having the suffix improves readability

Reply:
```suggestion
T_transfer_shard_id = TypeVar("T_transfer_shard_id")
T_partition_id = TypeVar("T_partition_id")
T_partition = TypeVar("T_partition")
```